### ISSUE TITLE
The cutwin package generates bad vertical spacings

### ISSUE THREAD

**Issue by hair-splitter** [davidcarlisle/cutwin#1]
I have a problem with the `cutwin` package. See the next MWE:
```
\documentclass[12pt]{article}
\usepackage{cutwin,lipsum}
\begin{document}
\opencutcenter
\begin{cutout}{4}{5cm}{5cm}{5}
\lipsum[1]
\end{cutout}
\end{document}
```
As you can see, the marked spacings are different:

![cutwin](https://user-images.githubusercontent.com/31892631/139673872-73406c0c-318e-485c-81df-d22056154ead.png)

**Comment by u-fischer** [comment]
Well yes, that is a quite old bug, you can find various variants if you search a bit. `cutwin` splits the paragraph into three boxes which it then outputs again. But its guesses about the needed glue between these boxes is often wrong. It is not so obvious in your example but also the spacing at the top of the window is not right. 

Your best bet is probably to set the two shifts cutwin uses to 0 *and* to add a \strut in various places:

~~~~
\documentclass[12pt]{article}
\usepackage{cutwin}

\ExplSyntaxOn
\newcommand\manyx[2]{\int_step_inline:nn{#1}{#2\c_space_tl}}
\ExplSyntaxOff
\begin{document}

\makeatletter
\renewcommand*{\c@twcalcilg}{\c@twvsilg=0pt }

\renewcommand*{\c@twcalcshift}{\c@twvsilg=0pt }
\makeatother


\section{with strut}

\begin{cutout}{4}{5cm}{5cm}{5}
\manyx{400}{\strut--}
\end{cutout}

\end{document}
~~~~

![image](https://user-images.githubusercontent.com/4047173/139701706-05792235-09d8-4489-a1da-e9f7814f3bf4.png)

**Comment by hair-splitter** [comment]
Thank you very much! Although this solution does not work correctly with normal text (or \lipsum[1]).

**Comment by u-fischer** [comment]
well no, it won't work with \lipsum. But with normal text it should be okay (but it requires a bit manual work to insert \strut in the relevant lines).

**Comment by hair-splitter** [comment]
Indeed, if I use \strut in the relevant lines, then the vertical spacings will be correct. Thanks!
